_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,265 @@
+
+![bundlephobia rtl screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/05.png)
+
+###### Source [Bundle Phobia](https://bundlephobia.com/package/@testing-library/react@12.1.2)
+
+
+Interesting facts: 
+- When you create a project by Create React App, you will get support for React Testing Library right away. 
+- Jest documentation links to the RTL website in the [*Testing Web Framework*](https://jestjs.io/docs/testing-frameworks#react) section.   
+
+## Basic differences
+
+The main differences between the two libraries that I can notice:
+1. **Different approaches to testing** 
+   
+   What distinguishes rtl is that we find elements by their labels or texts, almost like a user would. The point is to replicate and simulate the user's behavior. Rtl try to render components to the DOM and doesn’t give us access directly to the internal component. While Enzyme uses shallow rendering (or sometimes deep rendering) and encourages us to test the components instance using their state and props. Enzyme also allows full rendering of DOM but it is not the basic assumption of the library's operation.
+2. **Bundle size** 
+   
+   Maybe this is not as priority as the first point, but it is worth paying attention to bundle size also. RTL has a two times smaller bundle.
+3. **Age**  
+   
+   Enzyme 6 years old, RTL 3 years old
+
+## Comparison of downloads
+
+Below we can see a comparison of the number of downloads of both libraries in the last 5 years. Between August and September 2020, the number of RTL downloads exceeded the number of enzyme downloads.
+
+![enzyme and rtl dowloands graph](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/06.png)
+###### Source [npm trends](https://www.npmtrends.com/@testing-library/react-vs-enzyme)
+
+In the case of RTL, it is clear that the number of downloads is still growing strongly. On the other hand, it can be suspected that the highest peak of enzyme downloads is already behind it.
+
+###### Reasons to change testing library
+
+Well, let's be honest, Enzyme.js is a powerful library which helps us test our React Components in an easier way for a long time. But maybe.. too long? The first official release was in December 2015. Do not get me wrong, I don't want to say that everything old is bad. But the frontend world is changing constantly, just as the approach to testing it. 
+I know that the Enzyme.js library hasn’t been deprecated yet, but there are a lot of reasons to switch your test library right now. **If you haven't read it I highly recommend you to take a look at this [Piotr Staniów article](https://www.piotrstaniow.pl/goodbye-enzyme)** (it takes about 10 minutes to read). 
+
+The main reasons for me to think about slowly rewriting your tests and skip to another library from this article are: 
+- one maintains a developer, 
+- not keeping up with React changes. 
```

Comment:
there are other significant reasons to consider:
* concise API (few methods in RTL vs lots of them in Enzyme)
* one library, multiple frameworks - with the same API you can test angular, vue etc. using tesiting-library/angular